File: typescript/isogram/isogram.ts
export default class Isogram {

    static isIsogram(input: string): boolean {

        const sanitized: string = input.replace(/[- ]/, '').toUpperCase()
        let list: string[] = []
        let isogram = true

        Array.from(sanitized).forEach(letter => {
            if (list.includes(letter)) { isogram = false }
            list.push(letter)
        });

        return isogram
    }
}

File: csharp/isogram/Isogram.cs
using System.Collections.Generic;

public static class Isogram
{
    private static string _alphabet = "abcdefghijklmnopqrstuvwxyz";

    public static bool IsIsogram(string word)
    {
        word = word.ToLower();
        HashSet<char> hashset = new HashSet<char>();

        foreach (char letter in word)
        {
            if (_alphabet.Contains(letter))
            {
                if (!hashset.Add(letter))
                {
                    return false;
                }
            }
        }
        return true;
    }
}
